tests/visualization_tests/matplotlib_tests/test_edf.py

```diff
@@ -0,0 +1,31 @@
+import pytest
+
+from optuna.study import create_study
+from optuna.visualization.matplotlib import plot_edf
+
+
+@pytest.mark.parametrize("direction", ["minimize", "maximize"])
+def test_plot_optimization_history(direction: str) -> None:
+    # Test with no studies.
+    figure = plot_edf(create_study(direction=direction))
+    assert figure.has_data() is False
+
+    # Test with no trials.
+    figure = plot_edf(create_study(direction=direction))
+    assert figure.has_data() is False
+
+    figure = plot_edf([create_study(direction=direction), create_study(direction=direction)])
+    assert figure.has_data() is False
+
+    # Test with a study.
+    study0 = create_study(direction=direction)
+    study0.optimize(lambda t: t.suggest_float("x", 0, 5), n_trials=10)
+    figure = plot_edf(study0)
+    assert figure.has_data() is True
+
+    # Test with two studies.
+    # TODO(ytknzw): Add more specific assertion with the numbers of the studies.
+    study1 = create_study(direction=direction)
+    study1.optimize(lambda t: t.suggest_float("x", 0, 5), n_trials=10)
+    figure = plot_edf([study0, study1])
+    assert figure.has_data() is True
```

Comment:
This test code is completely the same as `tests/visualization_tests/test_edf.py`. How about using `pytest.mark.parametrize` to make the tests common? For example, you may be able to refer to `tests/samplers_tests/test_samplers.py`.

Reply:
I think having separate test codes is better than combining them, even though the cases should be the same.

1. Assertions could not be the same because the `figure` objects have different methods between Plotly and Matplotlib.
2. Having separate tests allows us to test Matplotlib backend only.

Reply:
Thanks for the comment. I agree with your claims. Let me resolve this conversation.